ledger/apply/heartbeat.go

```diff
@@ -105,11 +90,16 @@ func Heartbeat(hb transactions.HeartbeatTxnFields, header transactions.Header, b
 		return err
 	}
 	if hdr.Seed != hb.HbSeed {
-		return fmt.Errorf("provided seed %v does not match round %d's seed %v", hb.HbSeed, header.FirstValid, hdr.Seed)
+		return fmt.Errorf("provided seed %v does not match round %d's seed %v",
+			hb.HbSeed, header.FirstValid, hdr.Seed)
 	}
-
-	if !sv.Verify(id, hdr.Seed, hb.HbProof.ToOneTimeSignature()) {
-		return fmt.Errorf("heartbeat failed verification with VoteID %v", sv)
+	if account.VotingData.VoteID != hb.HbVoteID {
```

Comment:
there is no `account.VotingData.VoteID.IsEmpty` check so is possible to trick and call heartbeat on an account without voting keys? I suggest to have both match and empty checks

Reply:
Going into WellFormed.